Sources/OTLPGRPC/Metrics/OTLPGRPCMetricExporter.swift

```diff
@@ -0,0 +1,90 @@
+
+import GRPC
+import Logging
+import NIO
+import NIOHPACK
+import NIOSSL
+@_spi(Metrics) import OTel
+@_spi(Metrics) import OTLPCore
+
+/// Exports metrics to an OTel collector using OTLP/gRPC.
+@_spi(Metrics)
+public final class OTLPGRPCMetricExporter: OTelMetricExporter {
+    private let configuration: OTLPGRPCMetricExporterConfiguration
+    private let connection: ClientConnection
+    private let client: Opentelemetry_Proto_Collector_Metrics_V1_MetricsServiceAsyncClient
+    private let logger = Logger(label: String(describing: OTLPGRPCMetricExporter.self))
+
+    public init(
+        configuration: OTLPGRPCMetricExporterConfiguration,
+        group: any EventLoopGroup = MultiThreadedEventLoopGroup.singleton,
+        requestLogger: Logger = ._otelDisabled,
+        backgroundActivityLogger: Logger = ._otelDisabled
+    ) {
+        self.configuration = configuration
+
+        var connectionConfiguration = ClientConnection.Configuration.default(
+            target: .host(configuration.endpoint.host, port: configuration.endpoint.port),
+            eventLoopGroup: group
+        )
+
+        if configuration.endpoint.isInsecure {
+            logger.debug("Using insecure connection.", metadata: [
+                "host": "\(configuration.endpoint.host)",
+                "port": "\(configuration.endpoint.port)",
+            ])
+        }
+
+        // TODO: Support OTEL_EXPORTER_OTLP_CERTIFICATE
+        // TODO: Support OTEL_EXPORTER_OTLP_CLIENT_KEY
+        // TODO: Support OTEL_EXPORTER_OTLP_CLIENT_CERTIFICATE
+
+        var headers = configuration.headers
+        if !headers.isEmpty {
+            logger.trace("Configured custom request headers.", metadata: [
+                "keys": .array(headers.map { "\($0.name)" }),
+            ])
+        }
+        headers.replaceOrAdd(name: "user-agent", value: "OTel-OTLP-Exporter-Swift/\(OTelLibrary.version)")
+
+        connectionConfiguration.backgroundActivityLogger = backgroundActivityLogger
+        connection = ClientConnection(configuration: connectionConfiguration)
+
+        client = Opentelemetry_Proto_Collector_Metrics_V1_MetricsServiceAsyncClient(
+            channel: connection,
+            defaultCallOptions: .init(customMetadata: headers, logger: requestLogger)
+        )
+    }
+
+    public func export(_ batch: some Collection<OTelResourceMetrics> & Sendable) async throws {
+        if case .shutdown = connection.connectivity.state {
+            logger.error("Attempted to export batch while already being shut down.")
+            throw OTelMetricExporterAlreadyShutDownError()
+        }
+        let request = Opentelemetry_Proto_Collector_Metrics_V1_ExportMetricsServiceRequest.with { request in
+            request.resourceMetrics = batch.map(Opentelemetry_Proto_Metrics_V1_ResourceMetrics.init)
+        }
+
+        _ = try await client.export(request)
+    }
+
+    public func forceFlush() async throws {}
```

Comment:
```suggestion
    public func forceFlush() async throws {
        // This exporter is a "push exporter" and so the OTel spec says that force flush should do nothing.
    }
```

Do we want to add the comment here too?